feature/experimental/p4rt/tests/p4rt_election/p4rt_election_test.go

```diff
@@ -0,0 +1,625 @@
+package p4rt_election_test
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/cisco-open/go-p4/p4rt_client"
+	"github.com/cisco-open/go-p4/utils"
+	"github.com/openconfig/featureprofiles/feature/experimental/p4rt/internal/p4rtutils"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygot/ygot"
+	p4_v1 "github.com/p4lang/p4runtime/go/p4/v1"
+)
+
+// Flag variable definitions
+var (
+	p4InfoFile   = flag.String("p4info_file_location", "../../wbb.p4info.pb.txt", "Path to the p4info file.")
+	p4rtNodeName = flag.String("p4rt_node_name", "SwitchChip1/0", "component name for P4RT Node")
+)
+
+// Variable definitions
+var (
+	pDesc        = "Primary Connection"
+	sDesc        = "Secondary Connection"
+	npDesc       = "New Primary"
+	gdpEtherType = *ygot.Uint32(0x6007)
+	portId       = *ygot.Uint32(10)
+	deviceId     = *ygot.Uint64(1)
+	inId0        = *ygot.Uint64(0)
+	inId90       = *ygot.Uint64(90)
+	inId100      = *ygot.Uint64(100)
+	inId101      = *ygot.Uint64(101)
+	inId102      = *ygot.Uint64(102)
+	streamName   = "p4rt"
+)
+
+// Define PacketIO Interface
+type PacketIO interface {
+	GetTableEntry(delete bool) []*p4rtutils.ACLWbbIngressTableEntryInfo
+}
+
+// Define GDP Packet io structure
+type GDPPacketIO struct {
+	PacketIOPacket
+	IngressPort string
+}
+
+// Define PacketIO Structure
+type PacketIOPacket struct {
+	EthernetType *uint32
+}
+
+// Define Multiple Client Args Structure
+type mcTestArgs struct {
+	name     string
+	Items    [2]*testArgs
+	wantFail bool
+}
+
+// Define Test Args Structure
+type testArgs struct {
+	desc            string
+	lowID           uint64
+	highID          uint64
+	deviceID        uint64
+	handle          *p4rt_client.P4RTClient
+	wantStatus      int32
+	wantFinalStatus int32
+	wantFail        bool
+	wantWrite       bool
+	wantRead        bool
+}
+
+// configureDeviceId configures p4rt device-id on the DUT.
+func configureDeviceId(t *testing.T, dut *ondatra.DUTDevice) {
+	component := &oc.Component{}
+	component.IntegratedCircuit = &oc.Component_IntegratedCircuit{}
+	component.Name = ygot.String(*p4rtNodeName)
+	component.IntegratedCircuit.NodeId = ygot.Uint64(deviceId)
+	gnmi.Replace(t, dut, gnmi.OC().Component(*p4rtNodeName).Config(), component)
+}
+
+// configurePortId configures p4rt port-id on the DUT.
+func configurePortId(t *testing.T, dut *ondatra.DUTDevice) {
+	portName := dut.Port(t, "port1").Name()
+	gnmi.Replace(t, dut, gnmi.OC().Interface(portName).Id().Config(), portId)
+}
+
+// Create client connection
+func clientConnection(t *testing.T, dut *ondatra.DUTDevice) *p4rt_client.P4RTClient {
+	clientHandle := p4rt_client.P4RTClient{}
```

Comment:
Change to `p4rt_client.NewP4RTClient()` method. This would initialize `StreamTermErr`

Reply:
Thanks @Ankur19, changes worked. Just working on a few more things and should be in a decent place to push something for this week.